lib/TH/generic/THStorage.c

```diff
@@ -107,9 +107,9 @@ void THStorage_(free)(THStorage *storage)
   if(!storage)
     return;
 
-  if((storage->flag & TH_STORAGE_REFCOUNTED) && (storage->refcount > 0))
+  if((storage->flag & TH_STORAGE_REFCOUNTED) && (THAtomicGet(&storage->refcount) > 0))
   {
```

Comment:
Isn't this AtomicGet unnecessary? The ref counting for Tensors below just jumps straight to the DecrementRef.